reusable/images_to_numpy.py

```diff
@@ -0,0 +1,90 @@
+import os
+import cv2
+import numpy as np
+from tqdm import tqdm
+from PIL import Image
+import matplotlib.pyplot as plt
+
+# Assuming Folder Structure
+"""
+Dataset:
+    - Images
+        - image_01
+        - image_02
+        - image_03
+"""
+# Resize the image
+def crop_and_resize(img, resize_dim=1024):
+    img = cv2.resize(img, (resize_dim, resize_dim), interpolation=cv2.INTER_AREA)
+    return img
+
+# Read images and convert it to RGB
+def get_data(images):
+    img = cv2.imread(images)
+    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)  # Convert BGR to RGB
+    img = crop_and_resize(img)
+    return img
+
+# Load images and convert it to numpy
+def convert_to_numpy(path):
+    print(path)
+    folder_name = os.path.basename(path)  # Get the folder name
+    inp_feat = []
+    image_files = os.listdir(path)
+    for file_name in tqdm(image_files):
+        file_path = os.path.join(path, file_name)
+        img = get_data(file_path)
+        inp_feat.append(img)
+
+    inp_feat = np.array(inp_feat)
+    print(inp_feat.shape)
+    np.save(f"{folder_name}.npy", inp_feat)  # Save the .npy file with the folder name
+    print("Done!")
+
+def main() -> None:
+
+    data_path = "./Datasett/"
+
+    image_path = os.path.join(data_path, "Images")
+
+    convert_to_numpy(image_path)
+
+
+
+"""Optional"""
```

Comment:
take input from user whether to visualize outputs, then use the functions. Use [this](https://github.com/hasibzunair/plugnplay/blob/main/reusable/argparsers.py) for taking input from users.